src/Mollie/WC/Helper/ApplePayDataObjectHttp.php

```diff
@@ -0,0 +1,470 @@
+            filter_var_array(
+                $post['shippingContact'],
+                FILTER_SANITIZE_STRING
+            ),
+            self::SHIPPING_CONTACT_INVALID
+        );
+        $this->billingAddress = $this->completeAddress(
+            filter_var_array(
+                $post['billingContact'],
+                FILTER_SANITIZE_STRING
+            ),
+            self::BILLING_CONTACT_INVALID
+        );
+        $this->shippingMethod = filter_var_array(
+            $post['shippingMethod'],
+            FILTER_SANITIZE_STRING
+        );
+    }
+
+    /**
+     * Checks if the array contains all required fields and if those
+     * are not empty.
+     * If not it adds an unkown error to the object's error list, as this errors
+     * are not supported by ApplePay
+     * @param array $post
+     * @param array $required
+     *
+     * @return bool
+     */
+    protected function hasRequiredFieldsValues(array $post, array $required)
+    {
+        foreach ($required as $requiredField) {
+            if (!array_key_exists($requiredField, $post)) {
+                mollieWooCommerceDebug(
+                    "ApplePay Data Error: Missing index {$requiredField}"
+                );
+
+                array_push($this->errors, ['errorCode' => 'unknown']);
+                continue;
+            }
+            if (!$post[$requiredField]) {
+                mollieWooCommerceDebug(
+                    "ApplePay Data Error: Missing value for {$requiredField}"
+                );
+                array_push($this->errors, ['errorCode' => 'unknown']);
+                continue;
+            }
+        }
+        return !$this->hasErrors();
+    }
+
+    /**
+     * Sets the value to the appropriate field in the object
+     * @param array $post
+     */
+    protected function assignDataObjectValues(array $post)
+    {
+        foreach ($post as $key => $value) {
+            $filterType = $this->filterType($value);
+            $this->$key = $this->filterValue($value, $filterType);
+        }
+    }
+
+    /**
+     * Selector for the different filters to apply to each field
+     * @param $value
+     *
+     * @return int
+     */
+    protected function filterType($value)
+    {
+        $filterInt = ['productQuantity', 'productId'];
+        $filterBoolean = ['needShipping'];
+        switch ($value) {
+            case in_array($value, $filterInt):
+                return FILTER_SANITIZE_NUMBER_INT;
+                break;
+            case in_array($value, $filterBoolean):
+                return FILTER_VALIDATE_BOOLEAN;
+                break;
+            default:
+                return FILTER_SANITIZE_STRING;
+        }
+    }
+
+    /**
+     * Filter the input value with the given filter
+     * @param $value
+     * @param $filter
+     *
+     * @return mixed
+     */
+    protected function filterValue($value, $filter)
+    {
+        return filter_var($value, $filter);
+    }
```

Comment:
Does not sound redundant? why not just use `filter_var` instead of a method which is name ` filterVar`? You're gonna need a function stack everytime you call this method just to call `filter_var` which is already callable without any effort.